### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bohdan-mykhailenko**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -3,9 +3,26 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    const lastIndexOfElement = this.length - 1;
```
Comment (silchyn) [inline src/arrayMethodJoin.js]:
```suggestion
    const lastIndex = this.length - 1;
```

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -3,9 +3,26 @@
...
+    const emptyValues = [null, undefined];
```
Comment (silchyn) [inline src/arrayMethodJoin.js]:
```suggestion
    const ignoredValues = [null, undefined];
```

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -3,9 +3,26 @@
...
+
+    for (const element of this) {
```
Comment (silchyn) [inline src/arrayMethodJoin.js]:
```suggestion
    for (const character of this) {
```

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -3,9 +3,26 @@
...
+      const indexOfElement = this.indexOf(element);
+
+      if (!emptyValues.includes(element)) {
+        joinedString += element;
+      }
+
+      if (indexOfElement !== lastIndexOfElement) {
```
Comment (silchyn) [inline src/arrayMethodJoin.js]: and what if the initial string has duplicates?

**Review by silchyn — CHANGES_REQUESTED** [review]

**Review by naziamoff — APPROVED** [review]
awesome 🔥

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**